xarray/core/coordinates.py

```diff
@@ -236,13 +449,86 @@ def merge(self, other: Coordinates | None) -> Dataset:
             variables=coords, coord_names=coord_names, indexes=indexes
         )
 
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        self.update({key: value})
+
+    def update(self, other: Mapping[Any, Any]) -> None:
+        """Update this Coordinates variables with other coordinate variables."""
+        other_obj: Coordinates | Mapping[Hashable, Variable]
+
+        if isinstance(other, Coordinates):
+            # special case: default indexes won't be created
+            other_obj = other
+        else:
+            other_obj = getattr(other, "variables", other)
+
+        self._maybe_drop_multiindex_coords(set(other_obj))
+
+        coords, indexes = merge_coords(
+            [self.variables, other_obj],
+            priority_arg=1,
+            indexes=self.xindexes,
+        )
+
+        self._update_coords(coords, indexes)
+
+    def _overwrite_indexes(
+        self: T_Coordinates,
+        indexes: Mapping[Any, Index],
+        variables: Mapping[Any, Variable] | None = None,
+    ) -> T_Coordinates:
+        results = self.to_dataset()._overwrite_indexes(indexes, variables)
+        return cast(T_Coordinates, results.coords)
+
+    def _reindex_callback(
+        self: T_Coordinates,
```

Comment:
And here (not sure if it works with cast though..., Can always use a type: ignore)